typing/ctype.ml

```diff
@@ -1644,22 +1644,37 @@ let instance_label ~fixed lbl =
     (vars, ty_arg, ty_res)
   )
 
-let prim_mode mvar = function
-  | Primitive.Prim_global, _ -> Locality.allow_right Locality.global
-  | Primitive.Prim_local, _ -> Locality.allow_right Locality.local
+(* CR dkalinichenko: we must vary yieldingness together with locality to get
+   sane behavior around defaults. Remove once we have mode polymorphism. *)
+let prim_mode mvar mvar' = function
+  | Primitive.Prim_global, _ ->
+    Locality.allow_right Locality.global,
+    Yielding.allow_right Yielding.unyielding
+  | Primitive.Prim_local, _ ->
+    Locality.allow_right Locality.local,
+    Yielding.allow_right Yielding.yielding
```

Comment:
Is it meaningful to allow users to write `global yielding` or `local unyielding` on primitives?

Reply:
`local unyielding`, yes. For example, `%obj_dup` on a `local unyielding` value should also return a `local unyielding` value. `global yielding` I don't think makes much sense.

Reply:
After double-checking, I don't think we need that -- those constructors are only used in the middle-end, which doesn't care about `yielding`.